crates/sovereign-sdk/rollup-interface/src/fork/mod.rs

```diff
@@ -27,4 +186,153 @@ impl Fork {
+            v >= b'0' && v <= b'9'
+        }
+
+        // Find the index of colon
+        let mut i = 0;
+        let colon_idx = loop {
+            if i == bytes.len() {
+                return None;
+            }
+            if bytes[i] == b':' {
+                break i;
+            }
+            i += 1;
+        };
+
+        let (spec_id_bytes, height_bytes) = bytes.split_at(colon_idx);
+        // Ignore colon
+        let Some((_, height_bytes)) = height_bytes.split_first() else {
+            return None;
+        };
+
+        if spec_id_bytes.is_empty() || height_bytes.is_empty() {
+            return None;
+        }
+
+        // If multiple digits, first digit should not be 0
+        if (spec_id_bytes.len() > 1 && spec_id_bytes[0] == b'0')
+            || (height_bytes.len() > 1 && height_bytes[0] == b'0')
+        {
+            return None;
+        }
+
+        // Largest u8 is 3 digits, largest u64 is 20 digits
+        if spec_id_bytes.len() > 3 || height_bytes.len() > 20 {
+            return None;
+        }
+
+        // Parse spec_id
+        let mut i = 0;
+        let mut spec_id: u8 = 0;
+        while i < spec_id_bytes.len() {
+            if !is_digit(spec_id_bytes[i]) {
+                return None;
+            }
+
+            let digit = spec_id_bytes[i] - b'0';
+            let exp = spec_id_bytes.len() - i - 1;
+
+            let Some(multiplier) = 10u8.checked_pow(exp as u32) else {
+                return None;
+            };
+            let Some(to_add) = digit.checked_mul(multiplier) else {
+                return None;
+            };
+
+            if let Some(new_spec_id) = spec_id.checked_add(to_add) {
+                spec_id = new_spec_id;
+            } else {
+                return None;
+            }
+
+            i += 1;
+        }
+
+        let Some(spec_id) = SpecId::from_u8(spec_id) else {
+            return None;
+        };
+
+        // Parse activation_height
+        let mut i = 0;
+        let mut height: u64 = 0;
+        while i < height_bytes.len() {
+            if !is_digit(height_bytes[i]) {
+                return None;
+            }
+
+            let digit = (height_bytes[i] - b'0') as u64;
+            let exp = height_bytes.len() - i - 1;
+
+            let Some(multiplier) = 10u64.checked_pow(exp as u32) else {
+                return None;
+            };
+
+            let Some(to_add) = digit.checked_mul(multiplier) else {
+                return None;
+            };
+
+            if let Some(new_height) = height.checked_add(to_add) {
+                height = new_height;
+            } else {
+                return None;
+            }
+
+            i += 1;
+        }
+
+        Some(Fork {
+            spec_id,
+            activation_height: height,
+        })
+    }
+}
+
+/// Verifies the order of forks. `size` is needed here due to being in const environment,
+/// size of the fork might not be known beforehand.
```

Comment:
I first thought of having a CITREA_ENV environment variable, in which all the constant values decide their value on, which would allow us to hardcode the values for testnet, mainnet, local development etc. into the binary. This is also needed to be able to hardcode the forks as in your suggestion since forks will be environment dependent.

But after discussions with @eyusufatik, it seemed like having all these variables independently would give us maximum flexibility, which would allow us to see whether this flexibility could potentially be useful for some cases. I don't know if this is better, since too much flexibility is confusing and more complex to manage, but IMO it might be good to start like this, and over time we can narrow the usage down.